spec/jobs/problem_checks_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Jobs::ProblemChecks do
+  before do
+    AdminDashboardData.reset_problem_checks
+  end
+
+  after do
+    Discourse.redis.flushdb
+  end
+
+  it "runs the problem check that has been added and adds the messages to the load_found_scheduled_check_problems array" do
+    AdminDashboardData.add_scheduled_problem_check(:test_identifier) do
+      AdminDashboardData::Problem.maybe_create("big problem")
+    end
+
+    execute_job_class_with_args
+    problems = AdminDashboardData.load_found_scheduled_check_problems
+    expect(problems.count).to eq(1)
+    expect(problems.first).to be_a(AdminDashboardData::Problem)
+    expect(problems.first.to_s).to eq("big problem")
+  end
+
+  it "can handle the problem check returning multiple problems" do
+    AdminDashboardData.add_scheduled_problem_check(:test_identifier) do
+      [
+        AdminDashboardData::Problem.maybe_create("big problem"),
+        AdminDashboardData::Problem.maybe_create("yuge problem", priority: "high", identifier: "config_is_a_mess")
+      ]
+    end
+
+    execute_job_class_with_args
+    problems = AdminDashboardData.load_found_scheduled_check_problems
+    expect(problems.count).to eq(2)
```

Comment:
I think it'll be more accurate to test for the actual problems instead of just the count. There could be a bug in the future that ends up adding the same problem twice and this assertion will still pass.